src/web/lib/components/ThemeColorsEditor/index.js

```diff
@@ -2,28 +2,65 @@ import React from "react";
 import classnames from "classnames";
 import { SketchPicker } from "react-color";
 import onClickOutside from "react-onclickoutside";
-import { colorLabels, colorsWithAlpha } from "../../../../lib/constants";
+import { colorLabels, colorsWithAlpha, ESC } from "../../../../lib/constants";
 import { colorToCSS } from "../../../../lib/themes";
 import Metrics from "../../../../lib/metrics";
 
 import "./index.scss";
 
 class ThemeColorsEditor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Hmm, I don't think we need to add local state to this component. Basically `colorName` should be the same as `this.props.selectedColor` and `selected` should match `this.props.selectedColor !== null`. So we should be able to get the info needed just from the supplied props